examples/blurredbox/MainWindow.cpp

```diff
@@ -0,0 +1,205 @@
+        label->setText( QString( tr( "Blur Quality" ) + " ( %1 )" ).arg( value ) );
+    } );
+    slider->setValue( blurred->blurQuality() );
+}
+
+void MainWindow::createBlurSizeControls( BlurredBox* blurred, QskLinearBox* layout )
+{
+    auto* const label = new QskTextLabel( layout );
+    label->setTextColor( Qt::white );
+    label->setFontRole( QskSkin::MediumFont );
+    auto* const slider = new QskSlider( Qt::Horizontal, layout );
+    slider->setMinimum( 4.0 );
+    slider->setMaximum( 32.0 );
+    connect( slider, &QskSlider::valueChanged, [ blurred, label ]( qreal value ) {
+        blurred->setBlurSize( static_cast< float >( value ) );
+        label->setText( QString( tr( "Blur Size" ) + " ( %1 )" ).arg( value ) );
+    } );
+    slider->setValue( blurred->blurSize() );
+}
+
+void MainWindow::createBlurOpacityControls( BlurredBox* blurred, QskLinearBox* layout )
+{
+    auto* const label = new QskTextLabel( layout );
+    label->setTextColor( Qt::white );
+    label->setFontRole( QskSkin::MediumFont );
+    auto* const slider = new QskSlider( Qt::Horizontal, layout );
+    slider->setMinimum( 0.0 );
+    slider->setMaximum( 1.0 );
+    connect( slider, &QskSlider::valueChanged, [ blurred, label ]( qreal value ) {
+        blurred->setOpacity( value );
+        label->setText( QString( tr( "Blur Opacity" ) + " ( %1 )" ).arg( value ) );
+    } );
+    slider->setValue( blurred->opacity() );
+}
+
+void MainWindow::createBlurCornerRadiiControls( BlurredBox* blurred, QskLinearBox* layout )
+{
+    auto* const label = new QskTextLabel( layout );
+    label->setTextColor( Qt::white );
+    label->setFontRole( QskSkin::MediumFont );
+    auto* const bar = new QskSegmentedBar( Qt::Horizontal, layout );
+    enum
+    {
+        TL = 0,
+        TR = 1,
+        BR = 2,
+        BL = 3
+    };
+    bar->addText( "TL" );
+    bar->addText( "TR" );
+    bar->addText( "BR" );
+    bar->addText( "BL" );
+    auto* const slider = new QskSlider( Qt::Horizontal, layout );
+    slider->setMinimum( 0.0 );
+    slider->setMaximum( 80.0 );
+    connect( slider, &QskSlider::valueChanged, [ blurred, bar, label ]( qreal value ) {
+        auto shape = blurred->shape();
+        const auto format = tr( "Corner Radius" ) + " ( %1 )";
+
+        switch ( bar->currentIndex() )
+        {
+            case TL:
+                shape.setTopLeft( { value, value } );
+                break;
+            case TR:
+                shape.setTopRight( { value, value } );
+                break;
+            case BR:
+                shape.setBottomRight( { value, value } );
+                break;
+            case BL:
+                shape.setBottomLeft( { value, value } );
+                break;
+        }
+
+        label->setText( format.arg( value ) );
+        blurred->setShape( shape );
+        blurred->update();
+    } );
+    connect( bar, &QskSegmentedBar::currentIndexChanged, [ blurred, slider ]( int index ) {
```

Comment:
See https://github.com/KDE/clazy/blob/master/docs/checks/README-connect-3arg-lambda.md.

In this particular case the following call would be less error prone:
connect( bar, &QskSegmentedBar::currentIndexChanged, slider. ... );

However you would still have the lifecycle of blurred being unguarded. Lambdas as slots do have their pitfalls.

Reply:
Thanks for the hint. I activated the clazy check therefore and changed the third argument.